src/test/java/com/lucidworks/spark/SolrRelationTest.java

```diff
@@ -92,6 +106,51 @@ public void testFilterSupport() throws Exception {
     deleteCollection("testFilterSupport2");
   }
 
+  @Test
+  public void testNestedDataFrames() throws Exception {
```

Comment:
From what I understand, the preserve schema feature is to support storing ML models in Solr? if so, can you add a unit test that builds up a ML model, maybe something simple like a logistic regression and then store it into Solr and retrieve it back?